dart/lib/src/sentry_client.dart

```diff
@@ -140,18 +141,42 @@ class SentryClient {
       return event;
     }
 
-    if (event.exceptions?.isNotEmpty ?? false) return event;
+    if (event.exceptions?.isNotEmpty ?? false) {
+      return event;
+    }
+
+    final isolateName = getIsolateName();
+    // isolates have no id, so we use the hashCode as id
+    final isolateId = isolateName?.hashCode;
 
     if (event.throwableMechanism != null) {
-      final sentryException = _exceptionFactory.getSentryException(
+      var sentryException = _exceptionFactory.getSentryException(
         event.throwableMechanism,
         stackTrace: stackTrace,
       );
 
-      return event.copyWith(exceptions: [
-        ...(event.exceptions ?? []),
-        sentryException,
-      ]);
+      if (_options.platformChecker.isWeb) {
+        return event.copyWith(
+          exceptions: [
+            ...?event.exceptions,
+            sentryException,
+          ],
+        );
+      }
+
+      sentryException = sentryException.copyWith(threadId: isolateId);
+
+      final thread = SentryThread(
+        id: isolateId,
+        name: isolateName,
+        crashed: true,
+        current: true,
+      );
```

Comment:
I'd check if `isolateId` or `isolateName` are not null at least, otherwise it does not make sense I guess? 

Threads are only added if `options.attachThreads` is enabled as well, See the logic: https://github.com/getsentry/sentry-java/blob/896333a176c8d58cb85e33f8a10bf5c50ccce5c3/sentry/src/main/java/io/sentry/MainEventProcessor.java#L235-L243

I believe we should replicate it here.